src/client/Microsoft.Identity.Client.Desktop.WinUI3/WebView2WebUi/WinUI3WindowWithWebView2.cs

```diff
@@ -0,0 +1,403 @@
+            {
+                _logger.Error($"[WebView2Control] Redirection to url: {url} is not permitted - WebView2 will fail...");
+
+                _result = AuthorizationResult.FromStatus(
+                    AuthorizationStatus.ErrorHttp,
+                    MsalError.NonHttpsRedirectNotSupported,
+                    MsalErrorMessage.NonHttpsRedirectNotSupported);
+
+                readyToClose = true;
+            }
+
+            if (readyToClose)
+            {
+                // Complete the dialog with the result
+                _dialogCompletionSource?.TrySetResult(_result);
+
+                // Close the window
+                DispatcherQueue.TryEnqueue(() => this.Close());
+            }
+
+            return readyToClose;
+        }
+
+        private void WebView2_CoreWebView2Initialized(WebView2 sender, CoreWebView2InitializedEventArgs args)
+        {
+            if (args.Exception != null)
+            {
+                _logger.Error($"[WebView2Control] CoreWebView2 initialization failed: {args.Exception.Message}");
+                _dialogCompletionSource?.TrySetResult(
+                    AuthorizationResult.FromStatus(AuthorizationStatus.ErrorHttp));
+                return;
+            }
+
+            _logger.Info("[WebView2Control] CoreWebView2InitializationCompleted");
+
+            _webView2.CoreWebView2.Settings.AreDefaultContextMenusEnabled = false;
+            _webView2.CoreWebView2.Settings.AreDevToolsEnabled = false;
+            _webView2.CoreWebView2.Settings.AreHostObjectsAllowed = false;
+            _webView2.CoreWebView2.Settings.IsScriptEnabled = true;
+            _webView2.CoreWebView2.Settings.IsZoomControlEnabled = false;
+            _webView2.CoreWebView2.Settings.IsStatusBarEnabled = true;
+            _webView2.CoreWebView2.Settings.AreBrowserAcceleratorKeysEnabled = false;
+
+            // Handle title changes
+            if (_embeddedWebViewOptions.Title == null)
+            {
+                _webView2.CoreWebView2.DocumentTitleChanged += CoreWebView2_DocumentTitleChanged;
+            }
+            else
+            {
+                this.Title = _embeddedWebViewOptions.Title;
+            }
+
+            SetTitle(sender);
+        }
+
+        /// <summary>
+        /// Displays the authentication dialog and waits for the user to complete the authentication flow.
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation token to cancel the authentication operation.</param>
+        /// <returns>A task that represents the asynchronous authentication operation. The task result contains the authorization result.</returns>
+        public async Task<AuthorizationResult> DisplayDialogAndInterceptUriAsync(CancellationToken cancellationToken)
+        {
+            _cancellationToken = cancellationToken;
+            _dialogCompletionSource = new TaskCompletionSource<AuthorizationResult>();
+
+            // Register cancellation callback
```

Comment:
this cancels the operation when the dialog is closed. What if app developer passes a cancellation token to cancel the operation?